expression/constant_fold.go

```diff
@@ -23,13 +25,50 @@ func FoldConstant(expr Expression) Expression {
 	return e
 }
 
+// foldConstant for BuiltinIfSig like BuiltinIfIntSig, BuiltinIfRealSig, BuiltinIfTimeSig and so on.
+func foldConstantForBuiltinIfSig(args []Expression, ctx sessionctx.Context) (Expression, bool) {
```

Comment:
How about changing this function name to `foldScalarFuncForIF` and:
1. return the **folded** function if this `IF` function can be folded to a simper one
2. return the **origin** function if this `IF` function can not be folded.